File: library/src/main/java/com/zhuiyun/library/utils/NumberUtils.java
package com.zhuiyun.library.utils;

/**
 * @author: yun
 * @date: 2022/4/20 0020 09
 */
public class NumberUtils {
    public static int strToInt(String str){
        int result=0;
        try {
            result= (int) strToDouble(str);
        }catch (Exception e){
            return result;
        }
        return result;
    }


    public static double strToDouble(String str){
        double result=0.0;
        try {
            result=Double.parseDouble(str);
        }catch (Exception e){
            return result;
        }
        return result;
    }

    public static float strToFloat(String str){
        float result=0.0f;
        try {
            result=Float.parseFloat(str);
        }catch (Exception e){
            return result;
        }
        return result;
    }
}

File: library/src/main/java/com/zhuiyun/library/utils/NumberUtils.kt
package com.zhuiyun.library.utils

import com.zhuiyun.library.utils.NumberUtils
import java.lang.Exception

/**
 * @author: yun
 * @date: 2022/4/20 0020 09
 */
object NumberUtils {
    fun strToInt(str: String): Int {
        var result = 0
        result = try {
            strToDouble(str).toInt()
        } catch (e: Exception) {
            return result
        }
        return result
    }

    fun strToDouble(str: String): Double {
        var result = 0.0
        result = try {
            str.toDouble()
        } catch (e: Exception) {
            return result
        }
        return result
    }

    fun strToFloat(str: String): Float {
        var result = 0.0f
        result = try {
            str.toFloat()
        } catch (e: Exception) {
            return result
        }
        return result
    }
}